Compilation fails on NixOS

``` sh
../../../src/Game/File.c:1988:9: error: format not a string literal and no format arguments [-Werror=format-security]
         fprintf(file,str);
         ^~~~~~~
cc1: some warnings being treated as errors
```

libtool: 2.4.6
autoconf: 2.69
automake: 1.16.1
gnumake: 4.2.1
gcc: 7.4.0
yacc: 3.4.1
flex: 2.6.4
SDL: 2.0.9

--- a/Linux/shell.nix
+++ b/Linux/shell.nix
@@ -1,0 +1,14 @@
+{ pkgs ? import <nixpkgs> {} }:
+pkgs.mkShell {
+    buildInputs = [
+        pkgs.libtool
+        pkgs.autoconf
+        pkgs.automake
+        pkgs.gnumake
+        pkgs.gcc
+        pkgs.yacc
+        pkgs.flex
+        pkgs.SDL2
+    ];
+}
+

--- a/src/Game/File.c
+++ b/src/Game/File.c
@@ -1985,7 +1985,7 @@ void logfileLog(char *logfile, char *str)
 
     if ((file = fopen(logfile, "at")) != NULL)
     {
-        fprintf(file,str);
+        fprintf(file, "%s", str);
         fclose(file);
     }
 }

--- a/src/Game/GameChat.c
+++ b/src/Game/GameChat.c
@@ -648,7 +648,7 @@ void gcChatTextDraw(featom *atom, regionhandle region)
                     fontPrint(x,y,tpGameCreated.playerInfo[chat->playerindex].baseColor,temp);
                     x+=fontWidth(temp);
 
-                    sprintf(temp, strGetString(strWhisperedMessage));
+                    sprintf(temp, "%s", strGetString(strWhisperedMessage));
                     fontPrint(x,y,gcGameWhisperedColor, temp);
                     x+=fontWidth(temp);
 

--- a/src/Game/MultiplayerGame.c
+++ b/src/Game/MultiplayerGame.c
@@ -1272,7 +1272,7 @@ void mgProcessLimit(char *name, char *retmessage)
     else
     {
         mgPlayerLimit = 0;
-        sprintf(retmessage, strGetString(strCommandLimitOff));
+        sprintf(retmessage, "%s", strGetString(strCommandLimitOff));
     }
 
     return;

--- a/src/Game/Universe.c
+++ b/src/Game/Universe.c
@@ -5251,16 +5251,22 @@ void writeGameStatsToFile(char *filename)
                     fprintf(statsFile,"\n");
                     switch (universe.gameStats.playerStats[i].typeOfPlayerDeath)
                     {
-                        case PLAYERKILLED_DROPPEDOUT    : fprintf(statsFile,strGetString(strStatsDeathByDropout));                  break;
-                        case PLAYERKILLED_CAPTUREDSHIP  : fprintf(statsFile,strGetString(strStatsDeathByShipCapture));              break;
-                        case PLAYERKILLED_LOSTMISSION   : fprintf(statsFile,strGetString(strStatsDeathByLossOfMission));            break;
+                        case PLAYERKILLED_DROPPEDOUT:
+                            fprintf(statsFile, "%s", strGetString(strStatsDeathByDropout));
+                            break;
+                        case PLAYERKILLED_CAPTUREDSHIP:
+                            fprintf(statsFile, "%s", strGetString(strStatsDeathByShipCapture));
+                            break;
+                        case PLAYERKILLED_LOSTMISSION:
+                            fprintf(statsFile, "%s", strGetString(strStatsDeathByLossOfMission));
+                            break;
                     }
                 }
             }
             else
             {
                 fprintf(statsFile,"\n");
-                fprintf(statsFile,strGetString(strStatsDeathStillAlive));
+                fprintf(statsFile, "%s", strGetString(strStatsDeathStillAlive));
             }
 
             switch(tpGameCreated.startingResources)
@@ -5321,7 +5327,7 @@ void writeGameStatsToFile(char *filename)
                 if(verifyGameStatsShipsBounds(statsFile,i,universe.gameStats.playerStats[i].totalShips,1) == TRUE)
                     verifyGameStatsShipsAcquiredTotals(statsFile,i);
             fprintf(statsFile,"\n\n");
-            fprintf(statsFile,strGetString(strStatsTotalAcqShipsByType));
+            fprintf(statsFile, "%s", strGetString(strStatsTotalAcqShipsByType));
             fprintf(statsFile,"\n");
 
             for(j=0;j<TOTAL_NUM_SHIPS;j++)
@@ -5334,7 +5340,7 @@ void writeGameStatsToFile(char *filename)
                 }
             }
             fprintf(statsFile,"\n___________________\n\n\n");
-            fprintf(statsFile,strGetString(strStatsTotalAcqShipsByClass));
+            fprintf(statsFile, "%s", strGetString(strStatsTotalAcqShipsByClass));
             for(j=0;j<NUM_CLASSES;j++)
             {
                 if(universe.gameStats.playerStats[i].totalForEachClass[j] > 0)

--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -31,7 +31,7 @@ homeworld_LIBTOOLFLAGS = --preserve-dup-deps
 if PLATFORM_WIN32
 homeworld_LDADD = $(HW_LIBS) $(DEP_LIBS) @SDL_LIBS@ $(GL_LIBS) $(MOVIE_LIBS) $(NET_LIBS)
 else
-homeworld_LDADD = $(HW_LIBS) $(DEP_LIBS) -L/usr/X11R6/lib -lX11 -lXext @SDL_LIBS@ $(GL_LIBS) $(MOVIE_LIBS) $(NET_LIBS)
+homeworld_LDADD = $(HW_LIBS) $(DEP_LIBS) -lX11 @SDL_LIBS@ $(GL_LIBS) $(MOVIE_LIBS) $(NET_LIBS)
 endif
 homeworld_LDFLAGS = -Wl,--as-needed
 

--- a/src/SDL/main.c
+++ b/src/SDL/main.c
@@ -915,7 +915,7 @@ void DebugHelpDefault(char *string)
     }
 
     /*DialogBox(ghInstance, MAKEINTRESOURCE(IDD_CommandLine), NULL, CommandLineFunction);*/
-    printf(gHelpString);
+    printf("%s", gHelpString);
     free(gHelpString);                                       //done with string, free it
 }
 

